### PR TITLE
Add coverage, tests, documentation

### PR DESCRIPTION
**Pull request by danielmoi**
#### Description

This branch adds coverage, extends tests, alters naming, and adds documentation

----
#### Changes

- simplify / ruin naming of propTypes and components
- hook up NYC and istanbul
- hook up Travis, Coveralls
- extend readme
- add basic component tests

----
#### Testing Instructions

_Outline the steps to test or reproduce the PR._

----
#### Screenshots

_(if appropriate, they help the reviewer visualise the changes)_

----
#### Checklist:
- [x] Tests – Have you added or edited the test cases?
- [x] Rebase – Has this PR been rebased against `master`?

### CODE REVIEW

**Comment by anupvarghese** [comment]
LGTM 👍

File: .nyc_output/3c0b31381b67cfc1487b29dea8b6493d.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (anupvarghese) [inline .nyc_output/3c0b31381b67cfc1487b29dea8b6493d.json]: add these to gitignore & npmingnore

File: .nyc_output/3ffac3fbeb0842c25bb2cfcec66bef89.json [hunk 2]
```diff
@@ -0,0 +1 @@
...ary-expr","locations":[{"start":{"line":39,"column":55},"end":{"line":39,"column":71}},{"start":{"line":39,"column":75},"end":{"line":39,"column":101}}],"line":39},"8":{"loc":{"start":{"line":47,"column":56},"end":{"line":47,"column":103}},"type":"binary-expr","locations":[{"start":{"line":47,"column":56},"end":{"line":47,"column":73}},{"start":{"line":47,"column":77},"end":{"line":47,"column":103}}],"line":47},"9":{"loc":{"start":{"line":55,"column":56},"end":{"line":55,"column":103}},"type":"binary-expr","locations":[{"start":{"line":55,"column":56},"end":{"line":55,"column":73}},{"start":{"line":55,"column":77},"end":{"line":55,"column":103}}],"line":55}},"s":{"0":1,"1":1,"2":1,"3":1,"4":1,"5":1,"6":2,"7":4,"8":0,"9":4,"10":4,"11":1,"12":6,"13":6,"14":6,"15":1,"16":1,"17":0,"18":1,"19":1,"20":1,"21":1,"22":1,"23":1,"24":1,"25":1,"26":1,"27":1,"28":1,"29":1,"30":1,"31":1,"32":1},"f":{"0":2,"1":4,"2":6},"b":{"0":[2,0],"1":[2,2],"2":[0,4],"3":[6,0],"4":[6,6],"5":[1,1,1,1,1,1],"6":[0,1],"7":[1,1],"8":[1,1],"9":[1,1]},"_coverageSchema":"332fd63041d2c1bcb487cc26dd0d5f7d97098a6c","hash":"e2484c4ca13d2a618d1552a83c64e23136b04552","contentHash":"df8654f5743eeb0037ff9521df451184_10.2.0"}}
```
Comment (anupvarghese) [inline .nyc_output/3ffac3fbeb0842c25bb2cfcec66bef89.json]: this also

**Comment by anupvarghese** [comment]
Forgot to add this in the last PR, should we add a `form` key in `initialState` [here](https://github.com/danielmoi/react-select-multi/blob/coverage-and-tests/src/redux/reducer.js#L34)

File: .travis.yml [hunk 3]
```diff
@@ -2,10 +2,12 @@ language: node_js
 node_js:
   - "6"
 script:
-  - npm run build
+  # - npm run build
```
Comment (johndagostino) [inline .travis.yml]: we want the tests to actually run on travis right?

File: README.md [hunk 4]
```diff
@@ -1,3 +1,80 @@
...
+## Tests
+```
+yarn spec
+```
+
+
+----
+## Credits
+
+Thanks to [Lyn](https://github.com/lyntco), [JD](https://github.com/johndagostino), [Mike](https://github.com/Scoutski), [Anoop](https://github.com/anupvarghese), and [Logs](https://github.com/alexlogs) 🎉🎉🎉🎉🎉
```
Comment (johndagostino) [inline README.md]: remove 'alexlogs'

**Comment by lyntco** [comment]
LGTM

**Pull request merged**